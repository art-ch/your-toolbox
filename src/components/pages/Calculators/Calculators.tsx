'use client';

import { LuCalculator } from 'react-icons/lu';
import { NavigationContainer } from '../Navigation/components/NavigationContainer/NavigationContainer';
import { NavigationLinkCard } from '../Navigation/components/NavigationLinkCard/NavigationLinkCard';
import { PageHeading } from '@/components/PageHeading/PageHeading';
import { useTranslation } from 'react-i18next';

export const Calculators = () => {
  const { t } = useTranslation();

  return (
    <div>
      <PageHeading
        title={t('common:calculators')}
        subtitle={t('common:calculatorsDescription')}
      />
      <NavigationContainer>
        <NavigationLinkCard
          href="/calculators/ascetic-exercise-duration"
          icon={<LuCalculator />}
          title={t('asceticExerciseDuration:title')}
          subtitle={t('asceticExerciseDuration:subtitle')}
        />
      </NavigationContainer>
    </div>
  );
};
