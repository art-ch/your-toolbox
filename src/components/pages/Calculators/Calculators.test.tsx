import { render } from '@testing-library/react';

import { Calculators } from './Calculators';

describe('Calculators component', () => {
  it('should render correctly', () => {
    const { container } = render(<Calculators />);

    expect(container).toMatchSnapshot();
  });
});
