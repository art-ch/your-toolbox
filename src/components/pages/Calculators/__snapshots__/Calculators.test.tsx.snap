// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculators component should render correctly 1`] = `
<div>
  <div>
    <header
      class="text-center py-3"
    >
      <h1
        class="text-2xl md:text-4xl font-semibold leading-none tracking-tight"
      >
        Calculators
      </h1>
      <p
        class="mt-3"
      >
        Find all available calculators here
      </p>
    </header>
    <nav
      class="grid gap-5 md:grid-cols-2 lg:grid-cols-3 pt-1 pb-4 md:py-5"
    >
      <a
        href="/calculators/ascetic-exercise-duration"
      >
        <div
          class="rounded-lg border bg-card text-card-foreground shadow-xs min-h-full"
        >
          <div
            class="flex flex-col space-y-1.5 p-6"
          >
            <h3
              class="font-semibold tracking-tight flex gap-2 text-xl md:text-2xl"
            >
              <div
                class="mt-0.5"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    height="20"
                    rx="2"
                    width="16"
                    x="4"
                    y="2"
                  />
                  <line
                    x1="8"
                    x2="16"
                    y1="6"
                    y2="6"
                  />
                  <line
                    x1="16"
                    x2="16"
                    y1="14"
                    y2="18"
                  />
                  <path
                    d="M16 10h.01"
                  />
                  <path
                    d="M12 10h.01"
                  />
                  <path
                    d="M8 10h.01"
                  />
                  <path
                    d="M12 14h.01"
                  />
                  <path
                    d="M8 14h.01"
                  />
                  <path
                    d="M12 18h.01"
                  />
                  <path
                    d="M8 18h.01"
                  />
                </svg>
              </div>
              Ascetic Exercise Duration
            </h3>
          </div>
          <div
            class="p-6 pt-0"
          >
            Regular exercising boosts your well-being. Use this calculator to determine your workout duration.
          </div>
        </div>
      </a>
    </nav>
  </div>
</div>
`;
